main/src/ca/uwaterloo/flix/language/errors/LexerError.scala

```diff
@@ -16,5 +16,186 @@
 package ca.uwaterloo.flix.language.errors
 
 import ca.uwaterloo.flix.language.CompilationMessage
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.util.Formatter
+
+sealed trait LexerError extends CompilationMessage {
+  val kind = "Lexer Error"
+}
+
+object LexerError {
+
+  /**
+   * An error raised when an unexpected character, such as €, is encountered
```

Comment:
We write full sentences as documentation, so everything should end with a period.